azurerm/resource_arm_image.go

```diff
@@ -0,0 +1,417 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceArmImage() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmImageCreateUpdate,
+		Read:   resourceArmImageRead,
+		Update: resourceArmImageCreateUpdate,
+		Delete: resourceArmImageDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"source_virtual_machine_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"os_disk.os_type"},
+			},
+
+			"os_disk": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"os_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.Linux),
+								string(compute.Windows),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+
+						"os_state": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.Generalized),
+								string(compute.Specialized),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+
+						"managed_disk_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+
+						"blob_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
```

Comment:
I don't think this should be computed, if I understood the "Computed" property correctly. In the Developer's guide, "any Resource or Data Source property that should be stored in state but that the user should not be able to configure should have its  Computed  property set to  true."

Per API spec [here](https://docs.microsoft.com/en-us/rest/api/manageddisks/images/images-create) blobUri "specifies the URI in Azure storage of the blob that you want to use to create the virtual machine image" which indicates it should be changeable by the user.